src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -266,21 +269,29 @@ private static TextDocumentSyncOptions getTextDocumentSyncOptions() {
 
   @Override
   public CompletableFuture<Object> shutdown() {
-    return CompletableFutures.computeAsync(cancelToken -> {
-      cancelToken.checkCanceled();
-      securityHotspotsHandlerServer.shutdown();
-      analysisScheduler.shutdown();
-      bindingManager.shutdown();
-      telemetry.stop();
-      settingsManager.shutdown();
-      threadPool.shutdown();
-      httpClient.close();
-      serverNotifications.shutdown();
-      standaloneEngineManager.shutdown();
-      moduleEventsProcessor.shutdown();
-      branchManager.shutdown();
-      return new Object();
-    });
+    List.<Runnable>of(
```

Comment:
Previously this whole shutdown block was async (`CompletableFutures.computeAsync`), now we run it in the same thread, so it could retain the client while terminating ?